src/main/java/net/pms/network/RequestV2.java

```diff
@@ -799,51 +796,55 @@ private InputStream imageHandler(HttpResponse output) {
 
 		output.headers().set(HttpHeaders.Names.ACCEPT_RANGES, HttpHeaders.Values.BYTES);
 		output.headers().set(HttpHeaders.Names.CONNECTION, HttpHeaders.Values.KEEP_ALIVE);
-		output.headers().set(HttpHeaders.Names.EXPIRES, getFUTUREDATE() + " GMT");
+		output.headers().set(HttpHeaders.Names.EXPIRES, getFutureDate() + " GMT");
 		return getResourceInputStream(argument);
 	}
 
-	private String serverInfoHandler(HttpResponse output) throws IOException {
+	private String serverSpecHandler(HttpResponse output) throws IOException {
 		output.headers().set(HttpHeaders.Names.CACHE_CONTROL, HttpHeaders.Values.NO_CACHE);
 		output.headers().set(HttpHeaders.Names.EXPIRES, "0");
 		output.headers().set(HttpHeaders.Names.ACCEPT_RANGES, HttpHeaders.Values.BYTES);
 		output.headers().set(HttpHeaders.Names.CONNECTION, HttpHeaders.Values.KEEP_ALIVE);
-		inputStream = getResourceInputStream((argument.equals("description/fetch") ? "PMS.xml" : argument));
+		InputStream iStream = getResourceInputStream((argument.equals("description/fetch") ? "PMS.xml" : argument));
+
+		byte b[] = new byte[iStream.available()];
+		iStream.read(b);
+		String s = new String(b, StandardCharsets.UTF_8);
 
 		if (argument.equals("description/fetch")) {
-			byte b[] = new byte[inputStream.available()];
-			inputStream.read(b);
-			String s = new String(b, StandardCharsets.UTF_8);
-			s = s.replace("[uuid]", PMS.get().usn()); //.substring(0, PMS.get().usn().length()-2));
-
-			if (PMS.get().getServer().getHost() != null) {
-				s = s.replace("[host]", PMS.get().getServer().getHost());
-				s = s.replace("[port]", "" + PMS.get().getServer().getPort());
-			}
+			s = preparePmsSpec(s);
+		}
+		return s;
+	}
 
-			if (mediaRenderer.isXbox360()) {
-				LOGGER.debug("DLNA changes for Xbox 360");
-				s = s.replace("Universal Media Server", configuration.getServerDisplayName() + " : Windows Media Connect");
-				s = s.replace("<modelName>UMS</modelName>", "<modelName>Windows Media Connect</modelName>");
-				s = s.replace("<serviceList>", "<serviceList>" + CRLF + "<service>" + CRLF +
-					"<serviceType>urn:microsoft.com:service:X_MS_MediaReceiverRegistrar:1</serviceType>" + CRLF +
-					"<serviceId>urn:microsoft.com:serviceId:X_MS_MediaReceiverRegistrar</serviceId>" + CRLF +
-					"<SCPDURL>/upnp/mrr/scpd</SCPDURL>" + CRLF +
-					"<controlURL>/upnp/mrr/control</controlURL>" + CRLF +
-					"</service>" + CRLF);
-			} else {
-				s = s.replace("Universal Media Server", configuration.getServerDisplayName());
-				if (mediaRenderer.isSamsung()) {
-					// register UMS as a AllShare service and enable built-in resume functionality (bookmark) on Samsung devices
-					s = s.replace("<serialNumber/>", "<serialNumber/>" + CRLF
-							+ "<sec:ProductCap>smi,DCM10,getMediaInfo.sec,getCaptionInfo.sec</sec:ProductCap>" + CRLF
-							+ "<sec:X_ProductCap>smi,DCM10,getMediaInfo.sec,getCaptionInfo.sec</sec:X_ProductCap>");
-				}
+	private String preparePmsSpec(String pmsXml) {
```

Comment:
Not a big deal but it might make more sense for the new instances of `pms` to be `ums`

Reply:
@SubJunk yeah, I was not sure about that. I see that everywhere `pms` is used so I wanted to be consequent. But I agree that we should start use `ums` prefix where it is needed.

Currently I'm on business trip. Will be back home on Thursaday. Then I will change it.

Reply:
@SubJunk done